src/test/java/uk/gov/companieshouse/overseasentitiesapi/validation/RemoveValidatorTest.java

```diff
@@ -0,0 +1,78 @@
+package uk.gov.companieshouse.overseasentitiesapi.validation;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import uk.gov.companieshouse.overseasentitiesapi.mocks.RemoveMock;
+import uk.gov.companieshouse.overseasentitiesapi.model.dto.RemoveDto;
+import uk.gov.companieshouse.service.rest.err.Err;
+import uk.gov.companieshouse.service.rest.err.Errors;
+import uk.gov.companieshouse.overseasentitiesapi.validation.utils.ValidationMessages;
+import static uk.gov.companieshouse.overseasentitiesapi.model.dto.OverseasEntitySubmissionDto.REMOVE_FIELD;
+
+
+@ExtendWith(MockitoExtension.class)
+class RemoveValidatorTest {
+
+    private static final String LOGGING_CONTEXT = "12345";
+
+    private RemoveValidator removeValidator;
+    private RemoveDto removeDto;
+
+    @BeforeEach
+    public void init() {
+        removeValidator = new RemoveValidator();
+        removeDto = RemoveMock.getRemoveDto();
+    }
+    @Test
+    void testNoValidationErrorReportedWhenIsNotProprietorOfLandIsNull() {
+        removeDto.setIsNotProprietorOfLand(null);
+        Errors errors = removeValidator.validate(removeDto, new Errors(), LOGGING_CONTEXT);
+        assertFalse(errors.hasErrors());
+    }
+
+    @Test
+    void testNoValidationErrorReportedWhenIsNotProprietorOfLandIsNotNull() {
+        Errors errors = removeValidator.validate(removeDto, new Errors(), LOGGING_CONTEXT);
+        assertFalse(errors.hasErrors());
+    }
+
+    @Test
+    void testNoValidationErrorReportedWhenIsNotProprietorOfLandIsTrue() {
+        removeDto.setIsNotProprietorOfLand(true);
+        Errors errors = removeValidator.validate(removeDto, new Errors(), LOGGING_CONTEXT);
+        assertFalse(errors.hasErrors());
+    }
+
+    @Test
+    void testValidationErrorReportedWhenIsNotProprietorOfLandIsFalse() {
+        removeDto.setIsNotProprietorOfLand(false);
+        Errors errors = removeValidator.validate(removeDto, new Errors(), LOGGING_CONTEXT);
+        assertTrue(errors.hasErrors());
+    }
+
+    @Test
+    void testValidationErrorReportedWhenIsNotProprietorOfLandIsFalseTest() {
+        String qualifiedFieldName = getQualifiedFieldName(RemoveDto.IS_NOT_PROPRIETOR_OF_LAND_FIELD);
+        removeDto.setIsNotProprietorOfLand(false);
+        Errors errors = removeValidator.validate(removeDto, new Errors(), LOGGING_CONTEXT);
+        String validationMessage = ValidationMessages.NOT_NULL_ERROR_MESSAGE.replace("%s", qualifiedFieldName);
```

Comment:
This is the wrong validation message - it should be `ValidationMessages.NOT_VALID_ERROR_MESSAGE` to match what the validator uses. But I see that the test is still passing with this wrong message which indicates the checks being done in the `assertError()` method aren't thorough enough. I think that you copied this code from somewhere else, so I'll look in to how we fix this. Leave this one with me @HatiDev .

Reply:
Thanks Martin I had followed NOT_NULL_ERROR_MESSAGE =  test for null
NOT_EMPTY_ERROR_MESSAGE = test for true
NOT_VALID_ERROR_MESSAGE = test for false

Reply:
Ok. I now see what's happened. The message needs to be set to `ValidationMessages.NOT_VALID_ERROR_MESSAGE` here.

If you run the test it will fail but that's because line 72 needs to be changed to say `assertTrue`.